### PR TITLE
fix: update validation status and message when files exceed size limit

### PR DESCRIPTION
**Pull request by danieljhegeman**
- #3451 

### Reviewers
**Functional:** 
@Bento007 
**Readability:** 
@nayib-jose-gloria 
---

## Changes
Populate the `dataset_processing_status.validation_status` and `dataset_processing_status.validation_message` when a file is encountered that is too large for processing.

### CODE REVIEW

File: backend/corpora/common/upload.py [hunk 1]
```diff
@@ -52,9 +52,26 @@ def upload(
...
+        dataset = Dataset.get(db_session, dataset_id, collection_id=collection_id)
+        if not dataset:
+            raise NonExistentDatasetException(f"Dataset {dataset_id} does not exist")
+    else:
+        dataset = None
+
+    # Check if file size is within max size limit
     max_file_size_gb = CorporaConfig().upload_max_file_size_gb * GB
     if file_size is not None and file_size > max_file_size_gb:
-        raise MaxFileSizeExceededException(f"{url} exceeds the maximum allowed file size of {max_file_size_gb} Gb")
```
Comment (danieljhegeman) [inline backend/corpora/common/upload.py:57]: `url` is not needed, and I don't want to have to think through the implications of saving Collection / Dataset id's here (probably not an issue?) _as well as_ exposing the curator's google auth id (possibly an issue?)

**Comment by danieljhegeman** [comment]
@jahilton @brianraymor this fix will ensure that the curator can see that validation was invalid AND a descriptive message that it was due to file size if they retrieve the Collection using the API. 

It does _not_ include any sort of change to indicate the file size error on the UX, which would be more involved. I am assuming this is not needed, anyways.

File: backend/corpora/common/upload.py [hunk 2]
```diff
@@ -52,9 +52,26 @@ def upload(
...
+        error_message = f"The file exceeds the maximum allowed size of {max_file_size_gb} bytes"
```
Comment (Bento007) [inline backend/corpora/common/upload.py:66]:
```suggestion
        error_message = f"The file exceeds the maximum allowed size of {max_file_size_gb} Gb"
```

File: tests/unit/backend/corpora/dataset_processing/test_common.py [hunk 3]
```diff
@@ -206,3 +209,22 @@ def converter(_file):
...
+                collection.id,
+                "download_url_placeholder",
+                max_file_size + 1,  # Purport to exceed the max file size
+                "test_user",
+                dataset_id=dataset.id,
+            )
+        self.assertEqual(dataset.processing_status.validation_status, ValidationStatus.INVALID)
+        self.assertEqual(
+            dataset.processing_status.validation_message,
+            f"The file exceeds the maximum allowed size of {max_file_size} bytes",
```
Comment (Bento007) [inline tests/unit/backend/corpora/dataset_processing/test_common.py:229]:
```suggestion
            f"The file exceeds the maximum allowed size of {max_file_size} Gb",
```

**Review by Bento007 — CHANGES_REQUESTED** [review]
The size is in Gigabytes not bytes.

File: backend/corpora/common/upload.py (hunk 2 above)
Comment (danieljhegeman) [reply backend/corpora/common/upload.py]: sorry this had to be fixed above, because the GB multiplier is already being applied, thus making it bytes

**Review by Bento007 — APPROVED** [review]

**Comment by brianraymor** [comment]
> It does not include any sort of change to indicate the file size error on the UI for that Collection, which would be more involved. I am assuming this is not needed, anyways.

I think that's a good compromise. What is the UX experience? Silence? Validation failure with no details?

**Pull request merged**